### PR TITLE
Small test refactor

### PR DESCRIPTION
**Pull request by mellenburg**
-bump pytest
-split cluster out of conftest so it can be replaced outight
-remove pytest_dir option
-add helper for getting env options
-add helper for session tempfiles

### CODE REVIEW

**Comment by mellenburg** [comment]
@cmaloney This is all consolidated for ease of testing downstream. Please let me know what you would like to see split into other PRs

**Comment by cmaloney** [comment]
One PR looks fine for this

File: packages/pytest/buildinfo.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "sources": {
     "pytest": {
       "kind": "url_extract",
-      "url": "https://pypi.python.org/packages/f0/ee/6e2522c968339dca7d9abfd5e71312abeeb5ee902e09b4daf44f07b2f907/pytest-2.9.2.tar.gz",
-      "sha1": "586675a7ce39cd33e16b34cc539240b38c10d5e8"
+      "url": "https://pypi.python.org/packages/a0/2f/076c816e2402c4911ccee4b93ba0475145b7cffd0320ca8efa0add7c469c/pytest-3.0.3.tar.gz",
```
Comment (cmaloney) [inline packages/pytest/buildinfo.json]: :+1:

**Comment by mellenburg** [comment]
@cmaloney This is ready for review

File: test_util/test_runner.py [hunk 2]
```diff
@@ -66,16 +64,14 @@ def integration_test(
     test_boilerplate = """#!/bin/bash
 source /opt/mesosphere/environment.export
```
Comment (cmaloney) [inline test_util/test_runner.py]: can we just use dcos-shell instead nowadays? `dcos-shell <cmd>`?

Comment (cmaloney) [inline test_util/test_runner.py]: (maybe turn this just into the "env" setction. + `dcos-shell` call?)

**Comment by cmaloney** [comment]
Just noticed one thing: This should update `test_util/run-all` / the bash in the same way

**Pull request merged**